### PR TITLE
misc(v2): change blog front matter to snake_case

### PR DESCRIPTION
**Pull request by yangshun**
## Motivation

Standardize our front matter. Front matter should be snake_case 🐍. I'm preserving the camelCase for legacy reasons, so users using the camelCase will still work.

Also, I'm removing two following fields:

- `authorFBID` - This is really unnecessary and we don't need it. People should just provide the FB Graph API image URL if they wanted a FB profile picture
- `authorTwitter`- This was never ported to D2, but is still in the template and examples

I'm not updating the docs and template yet because it should be done only after alpha.35 is published or else there will be a mismatch between the doc and latest version.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Blog loads.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

### CODE REVIEW

File: packages/docusaurus-plugin-content-blog/src/blogUtils.ts [hunk 1]
```diff
@@ -1,3 +1,10 @@
+/**
```
Comment (yangshun) [inline packages/docusaurus-plugin-content-blog/src/blogUtils.ts]: Changes in this file are just nits.

File: packages/docusaurus-plugin-content-blog/src/blogUtils.ts [hunk 2]
```diff
@@ -11,7 +18,7 @@ export function truncate(fileString: string, truncateMarker: RegExp | string) {
 }
 
 // YYYY-MM-DD-{name}.mdx?
-// prefer named capture, but old node version do not support
+// prefer named capture, but old Node version does not support.
 const FILENAME_PATTERN = /^(\d{4}-\d{1,2}-\d{1,2})-?(.*?).mdx?$/;
 
 function toUrl({date, link}: DateLink) {
```
Comment (yangshun) [inline packages/docusaurus-plugin-content-blog/src/blogUtils.ts]: @endiliey why is there a hardcoded date 2019-01-01 here? On L53 as well o.o

Comment (endiliey) [reply packages/docusaurus-plugin-content-blog/src/blogUtils.ts]: Not me who put that. Iirc its hardcoded that way by some prevs contributor who added cjk support.

He wanted to make it clear that its yyyy-mm-dd and then take the length lol.

Comment (endiliey) [reply packages/docusaurus-plugin-content-blog/src/blogUtils.ts]: Maybe just substring(0,9) lol

**Comment by yangshun** [comment]
@endiliey done. Thanks for the tip

**Pull request merged**